### PR TITLE
[2.4] [Fix] Updated api bool fields with default=true to *bool, to be usable from client

### PR DESCRIPTION
**Pull request by rawmind0**
This PR updates api bool fields with default=true to *bool, to be usable from client:
- `IgnoreDockerVersion` on `rke_types`
- `UseInternalIPAddress` on `NodeCommonParams` and `IgnoreDaemonSets` on `NodeDrainInput` on `machine_types`
- `Compress` on `FluentForwarderConfig` on `logging_types`
- `TLS` on `SMTPConfig` on `alerting_types`

This is related with golang `Marshal` function from `encoding/json` that omits false bool if `omitempty` is set, https://github.com/golang/go/issues/13284#issuecomment-157288425

Fix issues https://github.com/terraform-providers/terraform-provider-rancher2/issues/41, https://github.com/terraform-providers/terraform-provider-rancher2/issues/264 and https://github.com/rancher/rancher/issues/25321

### CODE REVIEW

**Comment by mrajashree** [comment]
@rawmind0 this will need changes in rancher and rke because both repos use these fields as bool, so we can't merge this till the other two PRs are also ready for merging

**Comment by rawmind0** [comment]
Related to PRs rancher/rancher#27169 and rancher/rke#2064

**Comment by mrajashree** [comment]
@rawmind0 thanks, can we also test this doesn't break things on upgrade from 2.3 branch?

**Comment by rawmind0** [comment]
PR rebased to address conflicts. 

Upgrade has been tested successfully from rancher v2.3.6, to v2.4.2, to this changes on 2.4 branch.

File: go.mod [hunk 1]
```diff
@@ -6,6 +6,7 @@ replace (
 	github.com/knative/pkg => github.com/rancher/pkg v0.0.0-20190514055449-b30ab9de040e
 	github.com/matryer/moq => github.com/rancher/moq v0.0.0-20190404221404-ee5226d43009
 	k8s.io/client-go => k8s.io/client-go v0.18.0
+	launchpad.net/gocheck => github.com/go-check/check v0.0.0-20200227125254-8fa46927fb4f
```
Comment (dramich) [inline go.mod:9]: Why is this replace here?

Comment (rawmind0) [reply go.mod]: I was getting errors with this library updating go modules, but it seems not needed now. Removed

**Pull request merged**